src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -99,24 +156,111 @@ public void register(IEventBus bus)
     }
 
     /**
+     * Since the view is unmodifiable, can't cache it really. Gotta stream and collect each time
+     * This should not be a costly operation however, since the {@link RegistryObject} is cached.
+     * The modders can also cache the entries themselves.
      * @return The unmodifiable view of registered entries. Useful for bulk operations on all values.
      */
     public Collection<RegistryObject<T>> getEntries()
     {
+        if(entriesView.isEmpty())
+        {
+            return lazyEntriesView.stream().map(LazyRegistryObject::getRegistryObject).collect(Collectors.toSet());
```

Comment:
Given there is no caching going on and `lazyEntriesView` is not exposed externally anyways would it make more sense to remove `lazyEntriesView` all together and just do `lazyEntries.keySet()` (with the remainder of the line being as it currently is).

Reply:
Might be good yeah, though the goal is to have the set be unmodifiable so I added a Collectors.unmodifiable() to it
probably are better ways of doing this

Reply:
Wouldn't the unmodifiable part need to be done at the end of collecting them in a new set, and not the set you are streaming over?

Reply:
This is what i meant
`Collections.unmodifiableSet(lazyEntries.keySet().stream().map(LazyRegistryObject::getRegistryObject).collect(Collectors.toSet()))`
so yeah i do it at the end